File: app/src/main/java/com/thoughtworks/china/mobile/extension/DoubleExtension.java
package com.thoughtworks.china.mobile.extension;


public class DoubleExtension {

    public static double km(double value) {
        return value * 1000.0;
    }

    public static double m(double value) {
        return value;
    }

    public static double cm(double value) {
        return value / 100.0;
    }

    public static double mm(double value) {
        return value / 1000.0;
    }

    public static double ft(double value) {
        return value / 3.28084;
    }
}

File: app/src/main/kotlin/com/thoughtworks/china/mobile/extension/DoubleExtension.kt
package com.thoughtworks.china.mobile.extension


fun Double.km() = this * 1_000.0

fun Double.m() = this

fun Double.cm() = this / 100.0

fun Double.mm() = this / 1_000.0

fun Double.ft() = this / 3.28084

object Main {
    @JvmStatic
    fun main(args: Array<String>) {
        val value: Double = 2.0
        println("""
        |km: ${value.km()}
        |m: ${value.m()}
        |cm: ${value.cm()}
        |mm: ${value.mm()}
        |ft: ${value.ft()}
""".trimMargin())
    }
}